guevaraTorres/assignments/slidingwindow/lc53/lc53.js

```diff
@@ -1,7 +1,53 @@
+/*
+
+each window is size K
+conditon to determine if there's enough elements
+remove the first element in the subarray
+add the new included element in the sliding window
+
+variables:windowStart, windowEnd, maxSum, current
+
+edge cases: null or 0
+*/
+
+
+
+
 function max_subarray_size_k(k, arr) {
   // TODO: Write your code here
+
+  let maxSum = 0;
+  let windowStart = 0;
+  let current = 0;
+
+  if(arr.length == 0 || k > arr.length || k < 1) return 0;
+
+  for(let windowEnd = 0; windowEnd < arr.length; windowEnd++) {
+    current += arr[windowEnd];              //add windowEnd index value to current
+
+    if(windowEnd >= k-1) {                  //check to see if the window is the proper size
+    maxSum = Math.max(maxSum, current);     //find max sum of maxSum and current
+    current -= arr[windowStart];            //remove windowStart index value from current
+    windowStart++;                          //increment windowStart
+    }
+  }
+
+  return maxSum;
+
 }
```

Comment:
If it's nice, write it twice! The solution is fantastic but to build that muscle memory, write it out twice.

Reply:
Will do!